src/Workspaces/Core/Portable/Serialization/SerializerService.cs

```diff
@@ -174,6 +183,61 @@ public void Serialize(object value, ObjectWriter writer, CancellationToken cance
         }
     }
 
+    private static void Write(ObjectWriter writer, ImmutableDictionary<string, StructuredAnalyzerConfigOptions> optionsByLanguage)
+    {
+        writer.WriteCompressedUInt((uint)optionsByLanguage.Count);
+        foreach (var (language, options) in optionsByLanguage)
+        {
+            writer.WriteString(language);
+
+            foreach (var key in options.Keys)
```

Comment:
we need to sort this.  otherwise, we'll get different checksums for the same conceptual values.

Reply:
Makes sense!

Reply:
Why does the assertion mention SourceGeneratorExecutionVersionMap?

```
Assets only in new solution but not in incremental solution
p4p62QfrCRLnTkp6eoAGAQ==, Microsoft.CodeAnalysis.Serialization.SolutionCompilationStateChecksums
ebmmmsecAQ2G88ik6me9lQ==, SourceGeneratorExecutionVersionMap
    (ProjectId, #a94a356c-e92c-4596-b184-e7093f58b046 - TestProj): 0.0
```

Reply:
i don't know.  but i want to start with a def issue to begin with.  the SGEVM is def concerning, but haven't seen it prior to this PR... so trying to isolate variables.